src/store/sliced-async-storage.ts

```diff
@@ -55,57 +62,68 @@ const assertValueIsString = (value: unknown, key: string): value is string => {
 };
 
 export const SlicedAsyncStorage: Pick<AsyncStorageStatic, 'getItem' | 'setItem' | 'removeItem'> = {
-  getItem: async (key: string) => {
-    assertKeyIsString(key);
+  getItem: async (rootKey: string) => {
+    assertKeyIsString(rootKey);
 
-    const rootValue = await AsyncStorage.getItem(key);
+    const rootValue = await AsyncStorage.getItem(rootKey);
     const slicesCount = getValueSlicesCount(rootValue);
 
     if (slicesCount) {
-      const slicesKeys = range(0, slicesCount).map(sliceIndex => getSliceKey(key, sliceIndex));
+      const slicesKeys = buildSlicesKeys(rootKey, slicesCount);
       const slicesPairs = await AsyncStorage.multiGet(slicesKeys);
 
-      const result = slicesPairs.map(([, value]) => value ?? '').join('');
+      // (!) Don't use `Array.join('')` here - it is slower
+      const result = slicesPairs.reduce((acc, curr) => acc + curr ?? '', '');
 
       return result;
     }
 
     return rootValue;
   },
-  setItem: async (key: string, value: string) => {
-    assertKeyIsString(key);
-    assertValueIsString(value, key);
+
+  setItem: async (rootKey: string, value: string) => {
+    assertKeyIsString(rootKey);
+    assertValueIsString(value, rootKey);
+
+    const prevSlicesKeys = await readSlicesKeys(rootKey);
 
     if (value.length <= MAX_SLICE_LENGTH) {
-      const oldSlicesKeys = await getSlicesKeys(key);
-      await AsyncStorage.setItem(key, value);
-      await AsyncStorage.multiRemove(oldSlicesKeys);
+      await Promise.all([
+        AsyncStorage.setItem(rootKey, value),
+        // Removing possible leftovers
+        AsyncStorage.multiRemove(prevSlicesKeys)
+      ]);
     } else {
       const slicesCount = Math.ceil(value.length / MAX_SLICE_LENGTH);
-      const sliceSize = Math.ceil(value.length / slicesCount);
-      const slices = range(0, slicesCount).map(sliceIndex =>
-        value.slice(sliceIndex * sliceSize, (sliceIndex + 1) * sliceSize)
-      );
-      const prevSlicesKeys = await getSlicesKeys(key);
-      await AsyncStorage.multiSet([
-        [key, `${SLICED_ROOT_VALUE_PREFIX}${slicesCount}`],
-        ...slices.map((slice, index): [string, string] => [getSliceKey(key, index), slice])
+
+      const slices: StringRecord = {
+        [rootKey]: `${SLICED_ROOT_VALUE_PREFIX}${slicesCount}`
+      };
+      for (let i = 0; i < slicesCount; i++) {
+        const key = buildSliceKey(rootKey, i);
+        const val = value.slice(i * MAX_SLICE_LENGTH, (i + 1) * MAX_SLICE_LENGTH);
+        slices[key] = val;
```

Comment:
```suggestion
        slices[key] = value.slice(i * MAX_SLICE_LENGTH, (i + 1) * MAX_SLICE_LENGTH);
```
Creating a new copy of slice may take some time